test/hotspot/gtest/utilities/test_rbtree.cpp

```diff
@@ -300,6 +309,42 @@ struct ArrayAllocator {
     }
   }
 
+  void test_visit_outside_range() {
+    RBTreeInt rbtree;
+    using Node = RBTreeIntNode;
+
+    rbtree.upsert(2, 0);
+    rbtree.upsert(5, 0);
+
+    rbtree.visit_range_in_order(0, 0, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(0, 1, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(1, 1, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(3, 3, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(3, 4, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(4, 4, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(6,6, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(6, 7, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
+    rbtree.visit_range_in_order(7, 7, [&](const Node* x) {
+      EXPECT_TRUE(false) << "Range should not visit nodes";
+    });
```

Comment:
Use `FAIL()` instead of `EXPECT_TRUE`. Why not an array of test cases?

```c++
int test_cases[][] = {{0,1}, {1,1}};
for test_case in test_cases ...

```